migration/workspace-query-separation/src/main/java/ddf/migration/workspace/query/separation/WorkspaceQueryMigration.java

```diff
@@ -0,0 +1,215 @@
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.operation.Query;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.operation.impl.CreateRequestImpl;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.operation.impl.UpdateRequestImpl;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+import ddf.migration.api.DataMigration;
+import ddf.security.Subject;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.codice.ddf.security.common.Security;
+import org.opengis.filter.Filter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class WorkspaceQueryMigration implements DataMigration {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(WorkspaceQueryMigration.class);
+
+  private final CatalogFramework catalogFramework;
+
+  private final FilterBuilder filterBuilder;
+
+  private final InputTransformer xmlInputTransformer;
+
+  private static final Security SECURITY = Security.getInstance();
+
+  public WorkspaceQueryMigration(
+      CatalogFramework catalogFramework,
+      FilterBuilder filterBuilder,
+      InputTransformer xmlInputTransformer) {
+    this.catalogFramework = catalogFramework;
+    this.filterBuilder = filterBuilder;
+    this.xmlInputTransformer = xmlInputTransformer;
+  }
+
+  @Override
+  public void migrate() {
+    Subject systemSubject = SECURITY.runAsAdmin(SECURITY::getSystemSubject);
+
+    if (systemSubject == null) {
+      LOGGER.debug("An error occurred while attempting to run this migration as admin");
+      return;
+    }
+
+    systemSubject.execute(this::migrateWorkspaces);
+  }
+
+  private void migrateWorkspaces() {
+    LOGGER.info("Beginning workspace query data migration");
+    System.out.println("Started: Workspace Query Data Migration");
+
+    try {
+      Filter workspaceFilter =
+          filterBuilder.allOf(
+              filterBuilder.attribute(Core.METACARD_TAGS).like().text(WORKSPACE_TAG),
+              filterBuilder.attribute(WORKSPACE_QUERIES).is().like().text("*"));
+
+      Query query = new QueryImpl(workspaceFilter);
+      QueryRequest queryRequest = new QueryRequestImpl(query);
+
+      QueryResponse queryResponse = catalogFramework.query(queryRequest);
+
+      queryResponse
```

Comment:
I think we'd have to use `ResultIterable` to make sure all workspaces are migrated.